app.js

```diff
@@ -10,12 +11,35 @@ function addTask() {
   if (taskText !== '') {
     const taskItem = createTaskItem(taskText);
     todoList.appendChild(taskItem);
+
+    // we share the event with screen reader users in the live region.
+    const announcementText = taskText + " added";
+    announce(announcementText);
+
+
     todoInput.value = '';
   }
 }
 
+// to update the aria-live region
+function announce(text){
+   livePolite.textContent = text;
+}
+
+//generate a unique ID
+function createUniqueID(){
+   return "id-" + Math.floor(Math.random() * 100) + "-" + Date.now();
+}
+
 // Create new task item
 function createTaskItem(taskText) {
+
+  //to properly associate labels, aria-describedby etc. we need unique IDs on each item in the list.
+  //the simplest way to do this is to generate a unique ID.
+  const taskID = createUniqueID();
```

Comment:
Instead of using math.floor() to completely generate something random, wouldn't it be better if we had an id generated for each list item as in 1,2,3 for list item position or is this something that should be completely random?

Reply:
I chose random ID as if you delete items and then add another you can get duplicate IDs if you just use list position (unless you recreate all the IDs each time).

Just seemed the easier way to avoid conflicts in IDs...but anything works as long as IDs are unique.